webcompat/api/uploads.py

```diff
@@ -26,11 +23,71 @@
 uploads = Blueprint('uploads', __name__, url_prefix='/upload',
                     template_folder='../templates')
 JSON_MIME = 'application/json'
-images = UploadSet('uploads', IMAGES)
-configure_uploads(app, images)
+ALLOWED_FORMATS = ('jpg', 'jpeg', 'jpe', 'png', 'gif', 'bmp')
 
-# limit image uploads to 4MB
-patch_request_class(app, 4 * 1024 * 1024)
+
+class UploadNotAllowed(Exception):
+    """
+    This exception is raised when we try to save a file that is not
+    allowed. Allowed files are defined in ALLOWED_FORMATS
+    """
+
+
+class Upload(object):
+    '''Class that abstracts over saving image and screenshot uploads.
+
+    It performs a simple extension-based validation before saving to the
+    file system. If a file is not allowed, a UploadNotAllowed exception
+    is rased.
+    '''
+    def __init__(self, imagedata):
+        self.imagedata = imagedata
+        self.is_base_64 = (isinstance(imagedata, unicode) and
+                           imagedata.startswith('data:image/'))
+        self.is_filestorage = isinstance(imagedata, FileStorage)
+        self.file_ext = self.get_file_ext(imagedata)
+
+    def get_file_ext(self, imagedata):
+        '''Method to return the file extension.'''
+        if self.is_filestorage:
+            return imagedata.filename.split('.')[-1].lower()
+        if self.is_base_64:
+            # We only expect base64 png screenshot uploads,
+            # so we do a double check to verify that, and raise otherwise
+            # data:image/png;base64, <base64 string>
+            if imagedata.split('/')[1].split(';')[0] != 'png':
+                raise UploadNotAllowed()
+            return 'png'
```

Comment:
here I don't understand. Why do we allow only png. Will not it get into the way of attach screenshots in the comment and someone uploading a jpeg.